### DISCUSSION TITLE
🙏`fetch` api doesn't work in edge function

### DISCUSSION THREAD

**Discussion by gyhyfj** [vercel/community#4014] [Help]
Here is a [minimal reproduct](https://github.com/gyhyfj/vercel-edge-fetch-reproduct).
This function is deployed at https://test-peach-three.vercel.app/api/test.

In this reproduct, I try to fetch https://47.106.137.81:9001/api/test but failed in edge environment.

🙏 Can someone help me about this?

**Comment by amyegan** [comment]
Hey @gyhyfj. I don't understand what the problem is. I receive "hi" when I GET https://test-peach-three.vercel.app/api/test

Can you help me understand what's going wrong?

**Reply by gyhyfj** [reply]
The reproduct code is 
```ts[api/test.ts]
export const config = {
  runtime: "edge",
};

export default async function handler(request: Request) {
  const res = await fetch("https://47.106.137.81:9001/api/test");

  // try {
  //   console.log(await res.text()); // error code: 1003
  // } catch (e) {
  //   console.error("text() error", e);
  // }

  try {
    console.log(await res.json());
  } catch (e) {
    console.error("json() error", e); //json() error SyntaxError: Unexpected token 'e', "error code: 1003" is not valid JSON
  }

  return new Response("hi");
}
```
It does return "hi", but in the `try ... catch`, it shows that `await fetch("https://47.106.137.81:9001/api/test");` get `"error code: 1003"`.
I'm not sure if request on ip is not allowed in vercel edge function.

**Reply by amyegan** [reply]
Are you intending for `https://47.106.137.81:9001` to be the address of your Vercel project? Because of the dynamic nature of serverless deployments, they don't have a fixed IP address. You would need to use the project domain for this to work.

**Reply by gyhyfj** [reply]
No, `https://47.106.137.81:9001` is my server address, I can't fetch that in my Vercel project by `fetch("https://47.106.137.81:9001/api/test")`.
In my Vercel project, it fetch that address but get `error code 1003`

**Reply by amyegan** [reply]
The 1003 error means that direct IP access is not allowed for the request or the redirect location does not exist. Try using a domain instead of the IP address. 

https://stackoverflow.com/questions/18631488/what-does-http-response-code-1003-means

**Reply by xiaohuilam** [reply]
Could you please tell, why does vercel functions block request to fetch an external IP addresses?

**Reply by SamBallantyne** [reply]
I'm having the same issue: `Direct IP access is not allowed in Vercel's Edge environment`. To be clear, I'm trying to fetch an external resource. There's no domain name associated with it (nor does there need to be).

**Reply by SamBallantyne** [reply]
If this is a limitation of the edge runtime, it should probably be documented at https://vercel.com/docs/functions/limitations.

**Reply by nabildroid** [reply]
@SamBallantyne did you manage to find  a solution ?

**Reply by Bacis** [reply]
I get the same error: Direct IP access is not allowed in Vercel's Edge environment (hostname: 127.0.0.1)

**Comment by nabildroid** [comment]
i'm facing the same problem!

**Reply by Keristrasz** [reply]
@nabildroid me too, did you solve it?

**Reply by nabildroid** [reply]
End up configuring a domain for the API
